ql/src/java/org/apache/hadoop/hive/ql/exec/TableScanOperator.java

```diff
@@ -63,19 +63,19 @@
 
   private VectorizationContext taskVectorizationContext;
 
-  protected transient JobConf jc;
-  private transient boolean inputFileChanged = false;
+  protected JobConf jc;
```

Comment:
I actually tried not keeping these fields but I was running into all sorts of issues like unable to serialize/de-serialize or plan generating without metadata etc. 
I am not sure if we need to keep all of these fields or we can selectively choose, I went by almost all in interest of time. If Gopal or Rajesh thinks that this may cause performance issue I can open a follow-up to investigate and choose fields selectively.

Reply:
Let's create a follow-up to explore whether some of them may be made transient again and discuss over there.

Reply:
I have updated HIVE-24005 to investigate this.